### ISSUE TITLE
jruby - `find()` throws `StandardError: java.lang.Exception: unknown exception`

### ISSUE THREAD

**Issue by recontrapablo** [RaiMan/SikuliX1#188]
I'm fairly new to SikuliX but I successfully managed to get a running environment on a Windows 10 vm and be able to use sikuli from a ruby script using jruby. However, I'm trying to replicate the same environment on a win7 and calls to `find()` are throwing the following java exception:

```
C:\Users\qauser\Downloads>jruby test.rb

detected screen resolution = 1024x768
R[25,25 673x488]@S(0)
[log] highlight R[25,25 673x488]@S(0) for 1.0 secs
C:/jruby-9.2.8.0/lib/ruby/gems/shared/gems/sikulix-1.1.4.0/lib/sikulix/sikulix.rb:60: warning: constant ::NativeException is deprecated
StandardError: java.lang.Exception: unknown exception
                find at C:/jruby-9.2.8.0/lib/ruby/gems/shared/gems/sikulix-1.1.4.0/lib/sikulix/sikulix.rb:62
  <class:SikuliXTest> at test.rb:70
              <main> at test.rb:26
```

The code that is triggering the exception is 

```
...
pattern = Pattern.new(filename).similar(0.99)
match = app.window.find(pattern)
```

I'm comparing my 2 environments and one thing that caught my eye was that the contents of `%APPDATA%\SikuliX\SikulixLibs` are different, as `libtessearactxxx.dll` and `libeptlxxx.dll` are present in win10 and missing in win7. Could that be the root cause of the issue? (I did try to manually copy them but the issue persists)

Any pointers on how to obtain a more verbose version of the error that would allow me to make it work would be greatly appreciated :)

Environment details:
 * OS: Windows 7 Enterprise SP1
 * jruby: jruby 9.2.8.0 (2.5.3) 2019-08-12 a1ac7ff Java HotSpot(TM) 64-Bit Server VM 25.22
1-b11 on 1.8.0_221-b11 +jit [mswin32-x86_64]
* java: java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.221-b11, mixed mode)

**Comment by RaiMan** [comment]
I guess: SikuliX 1.1.4 only supports 64-Bit systems.

**Comment by RaiMan** [comment]
Sure, you are using the latest build of SikuliX 1.1.4?

With this version there should not be a tesserasct/lept dll in the SikulixLibs folder on Windows.

Just tested with the 1.1.4 api-jar on Win7-Pro-64: works.

**Comment by recontrapablo** [comment]
Thanks for the quick reply. I tried myself on a Win7 Enterprise SP1 running *on vmware workstation* and my ruby code indeed works. 

After that it stroke me what a more experienced SikuliX user would probably have spotted right away: the win7 I used on my original test is running *on a vsphere server*, and while I'm deploying Sikulix inside the vm guest, it seems the issue could be related to the fact that vsphere probably uses some kind of headless mode that is not suported by the opencv dll. Could that be the root cause of the issue? 

I guess that what confused me originally was:

 1. I was indeed able to launch an `App` and highlight its main window on the vpshere vm. Because of this, I assumed SikuliX installation was ok, and every other functionality was going to work
 2. the fact that the exception message was too generic 

I will try to find a way to make it run on my vsphere vm by going through the docs and forums. I hope I can still use SikuliX on that scenario.

**Comment by RaiMan** [comment]
SikuliX should show up an error, when it is working on a system, that is evaluated as headless by Java.
So since it is principally running, there might well be a special problem with the OpenCV stuff in this environment. The error comes later, since the setup of OpenCV support is deferred until the first time needed (searching an image in this case).
Since the JRuby support for SikuliX is not my first priority (due to lack of knowledge ;-) I would be interested about the details of your setup: What is where on your Windows system and how do you run a test. 
If you do not want to show it here: mail to sikulix---at---outlook---dot---com
Thanks.

**Comment by recontrapablo** [comment]
I apologize for not replying earlier to your last comment. I spent quite some time trying to compile a debug version of opencv342 so I could understand what was the issue until I finally decided to give a try on a Win10 vm running on vsphere. Much to my surprise, the same ruby script worked like a charm. 

So I guess the original issue is due to some misconfiguration on my win7 vm. I will look deeper into it, probably by re-installing the win7 vm from scratch and comparing it with the failing one.
